cli/flox-rust-sdk/src/models/manifest/composite.rs

```diff
@@ -0,0 +1,276 @@
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<ManifestBuild, MergeError> {
+        Ok(ManifestBuild::default())
+    }
+
+    /// TODO: Not implemented.
+    fn merge_containerize(
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<Option<ManifestContainerize>, MergeError> {
+        Ok(None)
+    }
+
+    fn merge(&self, manifest1: Manifest, manifest2: Manifest) -> Result<Manifest, MergeError> {
+        let manifest = Manifest {
+            version: Self::merge_version(&manifest1, &manifest2)?,
+            install: Self::merge_install(&manifest1, &manifest2)?,
+            vars: Self::merge_vars(&manifest1, &manifest2)?,
+            hook: Self::merge_hook(&manifest1, &manifest2)?,
+            profile: Self::merge_profile(&manifest1, &manifest2)?,
+            options: Self::merge_options(&manifest1, &manifest2)?,
+            services: Self::merge_services(&manifest1, &manifest2)?,
+            build: Self::merge_build(&manifest1, &manifest2)?,
+            containerize: Self::merge_containerize(&manifest1, &manifest2)?,
+        };
+
+        Ok(manifest)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::BTreeMap;
+
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+
+    #[test]
+    fn shallow_merger_no_deps() {
+        let composer = Manifest {
+            version: Version::<1>,
+            vars: ManifestVariables(BTreeMap::from([(
+                "composer-keep".to_string(),
+                "composer".to_string(),
+            )])),
+            ..Manifest::default()
+        };
+
+        let composite_manifest = CompositeManifest {
+            composer,
+            deps: vec![],
+        };
+
+        let merged = composite_manifest.merge(ShallowMerger).unwrap();
+
+        assert_eq!(merged, Manifest {
+            version: Version::<1>,
+            vars: ManifestVariables(BTreeMap::from([(
+                "composer-keep".to_string(),
+                "composer".to_string()
+            )])),
+            // TODO: Not implemented.
+            install: ManifestInstall::default(),
+            hook: ManifestHook::default(),
+            profile: ManifestProfile::default(),
+            options: ManifestOptions::default(),
+            services: ManifestServices::default(),
+            build: ManifestBuild::default(),
+            containerize: None,
+        })
+    }
+
+    #[test]
+    fn shallow_merger_with_deps() {
+        let dep1 = Manifest {
+            version: Version::<1>,
+            vars: ManifestVariables(BTreeMap::from([
+                ("dep1-keep".to_string(), "dep1".to_string()),
+                ("dep1-update".to_string(), "dep1".to_string()),
+            ])),
+            ..Manifest::default()
+        };
+
+        let dep2 = Manifest {
+            version: Version::<1>,
+            vars: ManifestVariables(BTreeMap::from([
+                ("dep1-update".to_string(), "dep2".to_string()),
```

Comment:
Suggestion: Having "dep1-update" _also_ be "dep2" makes the intent a little unclear.

Reply:
The intent is that the key shows where value was originally set and the value shows where it was last set. Would changing value to be "updated by dep2" be clearer?

Reply:
Oh I didn't get that at all lol Yes, that would by clearer.

Reply:
Done (will squash)